src/main/java/uk/gov/ons/ctp/common/rabbit/RabbitHelper.java

```diff
@@ -0,0 +1,279 @@
+package uk.gov.ons.ctp.common.rabbit;
+
+import com.godaddy.logging.Logger;
+import com.godaddy.logging.LoggerFactory;
+import com.rabbitmq.client.AMQP.Queue.PurgeOk;
+import com.rabbitmq.client.Channel;
+import com.rabbitmq.client.Connection;
+import com.rabbitmq.client.ConnectionFactory;
+import com.rabbitmq.client.GetResponse;
+import java.io.IOException;
+import java.util.concurrent.TimeoutException;
+import uk.gov.ons.ctp.common.error.CTPException;
+import uk.gov.ons.ctp.common.error.CTPException.Fault;
+import uk.gov.ons.ctp.common.event.EventPublisher;
+import uk.gov.ons.ctp.common.event.EventPublisher.EventType;
+import uk.gov.ons.ctp.common.event.EventPublisher.RoutingKey;
+import uk.gov.ons.ctp.common.event.EventPublisher.Source;
+import uk.gov.ons.ctp.common.event.NativeRabbitEventSender;
+import uk.gov.ons.ctp.common.event.RabbitConnectionDetails;
+import uk.gov.ons.ctp.common.event.model.EventPayload;
+
+/**
+ * This is a test support class for interacting with RabbitMQ.
+ *
+ * <p>It runs as a singleton and the connection is established on first usage. When connecting to
+ * Rabbit it uses the connection details from a property file, with any or all of these fields
+ * overridable with equivalent environment variables.
+ *
+ * <p>The RabbitMQ Java API does not support concurrent usage of the Channel object, so this class
+ * enforces this restriction with method level synchronisation.
+ */
+public class RabbitHelper {
+  private static final Logger log = LoggerFactory.getLogger(RabbitHelper.class);
+
+  private static RabbitHelper rabbitInteraction = null;
+
+  private Connection rabbit;
+  private Channel channel;
+  private String exchange;
+
+  private EventPublisher eventPublisher;
+
+  private RabbitHelper(RabbitConnectionDetails rabbitDetails, String exchange) throws CTPException {
+    ConnectionFactory factory = new ConnectionFactory();
```

Comment:
Please make sure that when you add the yml and environment var overloading mech, that it is not simply burned into this code - rather make that mech reusable in common-service

Reply:
ok.